pkg/p2p/libp2p/libp2p.go

```diff
@@ -249,8 +249,7 @@ func (s *Service) AddProtocol(p p2p.ProtocolSpec) (err error) {
 			peerID := streamlibp2p.Conn().RemotePeer()
 			overlay, found := s.peers.overlay(peerID)
 			if !found {
-				// todo: this should never happen, should we disconnect in this case?
-				// todo: test connection close and refactor
+				// todo: this should never happen
```

Comment:
this `todo` really scares me. can this happen?

Reply:
Well it should not, as the system is designed that overlay sohuld be there as well at this point, but still I left the check to gurad againist it. Maybe I should remove the comment as well, as it is a git misleading.